### PR TITLE
Feature/brand yaml schema consolidation

### PR DESCRIPTION
**Pull request by cscheid**
Towards https://github.com/quarto-dev/quarto-cli/issues/10249.

This PR contributes what's (aspirationally) the schema that will ship with the 1.0 version of the spec.

### CODE REVIEW

File: tests/docs/smoke-all/typst/brand-yaml/logo/posit/brand-logo.qmd [hunk 1]
```diff
@@ -9,7 +9,7 @@ _quarto:
     typst:
       ensureTypstFileRegexMatches:
       -
-        - '#set page\(background: align\(left\+top, box\(inset: 0.5in, image\("posit-logo-2024.svg", width: 2in\)\)\)\)'
+        - '#set page\(background: align\(left\+top, box\(inset: 0.5in, image\("posit-logo-2024.svg", width: 2in, alt: "Posit Logo"\)\)\)\)'
```
Comment (cscheid) [inline tests/docs/smoke-all/typst/brand-yaml/logo/posit/brand-logo.qmd:12]: Huh, I didn't know alt text was supported in Typst! Very cool.

**Review by cscheid — COMMENTED** [review]
I didn't know about alt text in Typst - neat.

**Comment by gordonwoodhull** [comment]
I've updated the Typst implementation.

Could probably use more tests, and I think `brand.ts` should prepend `brandDir` on paths it returns. Run out of time for the moment, will try that later.

Note that with the current implementation we can directly use an image path as a string under `small`, `medium`, or `large`, but we can't provide alt text there - only in `images`. I'm not sure if it this is intentional.

I can't tell what the alt text does in a PDF, not getting hover in Preview.

**Comment by cscheid** [comment]
> I think `brand.ts` should prepend `brandDir` on paths it returns. Run out of time for the moment, will try that later.

That's almost certainly true. I thought I had tested that for typography with system files, but I think you're talking about `logo` here.

In any case, I echo the sentiment that `Brand` should resolve its paths to become absolute.

> Note that with the current implementation we can directly use an image path as a string under `small`, `medium`, or `large`, but we can't provide alt text there - only in `images`. I'm not sure if it this is intentional.

It wasn't intentional, but since there's some "path vs named image" ambiguity in the schema which we won't be able to address, I think we can leave that for now. A fix would be to change `brand-string-light-dark`:

```yml
- id: brand-string-light-dark
  anyOf:
    - string # we could change this to `ref: brand-logo-resource`
    - object:
        closed: true
        properties:
          light:
            schema: string
            description: >
              A link or path to the brand's light-colored logo or icon.
          dark:
            schema: string
            description: >
              A link or path to the brand's dark-colored logo or icon.
```

> I can't tell what the alt text does in a PDF, not getting hover in Preview.

Huh. I wonder if they snuck that in only to "secretly" use it in their HTML targets. I know they use an unpublished tool to create their docs, and note how https://typst.app/docs/tutorial/writing-in-typst/ has alt text in it:

image

**Pull request merged**